src/coins.cpp

```diff
@@ -152,6 +152,55 @@ CCoinsModifier CCoinsViewCache::ModifyNewCoins(const uint256 &txid, bool coinbas
     return CCoinsModifier(*this, ret.first, 0);
 }
 
+void CCoinsViewCache::AddCoin(const COutPoint &outpoint, CCoin&& coin, bool possible_overwrite) {
+    assert(!coin.IsPruned());
+    auto ret = cacheCoins.emplace(std::piecewise_construct, std::forward_as_tuple(outpoint.hash), std::forward_as_tuple(nullptr));
+    bool fresh = false;
+    if (!ret.second) {
+        cachedCoinsUsage -= ret.first->second.coins.DynamicMemoryUsage();
+    }
+    if (!possible_overwrite) {
+        if (ret.first->second.coins.IsAvailable(outpoint.n)) {
+            throw std::logic_error("Adding new coin that replaces non-pruned entry");
+        }
+        fresh = !(ret.first->second.flags & CCoinsCacheEntry::DIRTY);
```

Comment:
Agree, but again I'd rather not rewrite history. I can fix it when squashing, but just always setting it to false would not work (it would break the tests, which are very useful in determining that the correct behaviour remains during the whole changeset).